onyo/lib/onyo.py

```diff
@@ -38,57 +38,129 @@ def __init__(self, path: Union[Path, str], init: bool = False) -> None:
+        Onyo repository.
+
+        The cache will be consistent with the state of the repository as long as
+        only the `Repo`s public functions are used. Use of private functions
+        might require a manual reset of the caches with `Repo.clear_caches()`.
+        """
         if not self._files:
             self._files = self._get_files()
 
         return self._files
 
     @property
     def files_changed(self) -> set[Path]:
+        """
+        Returns a `set` containing the `Path`s relative to repo.root to all
+        changed files (according to git) of an Onyo repository.
+        """
         return self._get_files_changed()
 
     @property
     def files_staged(self) -> set[Path]:
+        """
+        Returns a `set` containing the `Path`s relative to repo.root to all
+        staged files (according to git) of an Onyo repository.
+        """
         return self._get_files_staged()
 
     @property
     def files_untracked(self) -> set[Path]:
+        """
+        Returns a `set` containing the `Path`s relative to repo.root to all
+        untracked files (according to git) of an Onyo repository
+        """
         return self._get_files_untracked()
 
     @property
     def opdir(self) -> Path:
+        """
+        Returns the absolute `Path` to the working directory where the
+        `Repo` object was instantiated.
+        """
         return self._opdir
 
     @property
     def root(self) -> Path:
+        """
+        Returns the absolute `Path` to the root of the `Repo` where the
+        directories `.git/` and `.onyo/` are located.
+        """
         return self._root
 
     @property
     def templates(self) -> set[Path]:
+        """
+        A `set` containing the `Path`s relative to repo.root to all
+        template files of an Onyo repository.
+
+        The cache will be consistent with the state of the repository as long as
+        only the `Repo`s public functions are used. Use of private functions
+        might require a manual reset of the caches with `Repo.clear_caches()`.
+        """
         if not self._templates:
             self._templates = self._get_templates()
 
         return self._templates
 
-    def clear_caches(self) -> None:
-        self._assets = None
-        self._dirs = None
-        self._files = None
-        self._templates = None
+    def clear_caches(self, assets: bool = True, dirs: bool = True,
+                     files: bool = True, templates: bool = True) -> None:
+        """
+        Clear all specified caches of the repository class.
+
+        The class saves paths, which can get invalidated when files or
+        directories are modified. To force onyo to reload the paths, this
+        function clears the caches.
+
+        By default all caches are cleared, and flags make it possible to specify
+        which caches which should remain set.
+
+        If the repository is only modified via the public API functions, the
```

Comment:
> If the repository is only modified via the public API functions

to

> If the repository is modified via public API functions

Reply:
I think then it would not be true anymore. If you use the Repo "via public API functions", the sentence does not exclude the usage of other functions. And they must in my opinion by specifically excluded, because not all public API functions do reset the cache, so the usage of public API functions does not keep them consistent, except when *only* using them.

Reply:
Yes. I agree. :-)